geopandas/geodataframe.py

```diff
@@ -1620,13 +1673,20 @@ def explode(self, column=None, **kwargs):
 
         # reset to MultiIndex, otherwise df index is only first level of
         # exploded GeoSeries index.
-        df.set_index(exploded_index, append=True, inplace=True)
-        df.index.names = list(self.index.names) + [None]
+        df.set_index(exploded_index, inplace=True)
 
-        if "__level_1" in df.columns:
-            df = df.rename(columns={"__level_1": "level_1"})
+        if add_multiindex:
+            df.index.names = list(self.index.names) + [None]
+        else:
+            df.index.names = self.index.names
+        if f"__{level_str}" in df.columns:
+            df = df.rename(columns={f"__{level_str}": level_str})
 
         geo_df = df.set_geometry(self._geometry_column_name)
+
+        if ignore_index:
+            geo_df.reset_index(inplace=True, drop=True)
```

Comment:
We can probably move this check a bit up, so we can avoid the `set_index` in this case (since we are just resetting it anyway)